docs/blog/2025-01-20-linking-ts-in-monorepos.md

```diff
@@ -0,0 +1,593 @@
+---
+title: 'Linking TypeScript in Monorepos'
+slug: linking-ts-in-monorepos
+authors: [Juri Strumpflohner]
+tags: []
+---
+
+Managing TypeScript packages in a monorepo comes with unique challenges. As your projects grow, so does the complexity of structuring, linking, and resolving dependencies between packages. From using simple relative imports to leveraging TypeScript path aliases, project references, and even modern workspace protocols, developers have a variety of strategies at their disposal. But which one is the right fit for your needs?
+
+{% toc /%}
+
+## What Does "Linking Packages" Actually Mean?
+
+When you work in a monorepo, the goal is to split logic into separate packages. Why? To create smaller, self-contained, and maintainable units. This approach not only enhances reusability but also helps scale—whether that's scaling teams or optimizing CI pipelines.
+
+As you split logic into packages, you'll inevitably need to link them together. At the **code level**, this is typically expressed through an import statement on the consumer side—whether that's an app using a package or one package depending on another.
+
+```ts
+import { something } from '@tsmono/mypackage';
+```
+
+To make this work, there are two key phases:
+
+- **Building:** This includes **type checking** and optionally **compilation/transpilation**.
+- **Execution:** This is when the app runs.
+
+TypeScript handles type checking and optionally compilation, while tools like esbuild, Rspack, or Vite ensure paths are resolved when running the app. In this article, we'll mostly focus on the "building" phase.
```

Comment:
It might be me, but I think it's a bit unclear what we're referring to with "compilation/transpilation" being optional. At first, I thought we were referring to that not being needed in platforms like Deno or runtimes like Bun, but I think the intention here is to say that we don't need to use `tsc` to "compile/transpile" because it can be done by other build tools, is that the intention?

If that's the case, maybe we should be clearer about that. Note we don't explicitly mention compile/transpile with `tsc`, and what we say is that compiling/transpiling is optional.

Reply:
yeah the thought was not doing the compilation/transpilation with `tsc` specifically but with tools like Vite etc (which handle it internally) at least from the user's perspective. But I see the confusion. Going to fix it